internal/controller/vrg_volsync.go

```diff
@@ -142,13 +142,13 @@ func (v *VRGInstance) reconcilePVCAsVolSyncPrimary(pvc corev1.PersistentVolumeCl
 
 	protectedPVC := v.findProtectedPVC(pvc.Namespace, pvc.Name)
 	if protectedPVC == nil {
-		protectedPVC = newProtectedPVC
-		v.instance.Status.ProtectedPVCs = append(v.instance.Status.ProtectedPVCs, *protectedPVC)
+		protectedPVC = v.addProtectedPVC(newProtectedPVC.Namespace, newProtectedPVC.Name)
 	} else if !reflect.DeepEqual(protectedPVC, newProtectedPVC) {
 		newProtectedPVC.Conditions = protectedPVC.Conditions
```

Comment:
Old code updated the conditions on newProtectedPVC - but this does not update the condition on the copy in the instance - looks like a bug.

Reply:
Not a bug

Reply:
Yes, that's a bug. The old code updated a copy. It fixes itself, however, in the next reconciliation given that the condition is already there.